content/architecture/use-modules-to-define-features.md

```diff
@@ -0,0 +1,114 @@
+---
+title: use modules to define features
+author: 
+  name: Billy Lando
+  url: https://github.com/billyjov
+---
+
+# Problem 
+
+As an app scales, the root module start growing. Loading multiple components increase the start time and affect the application performance.    
+The more components added, the more confusing it will be to maintain the entire application.
+
+# Solution 
+
+Module in Angular is a mechanism to group and encapsulate components, services, directives and pipes.   
+Using module to define features in Angular allows lazy loading, isolation and portability of this features. Feature modules are designed to be used accross multiple components and and their UI will be hosted in a parent Module e.g `AppModule`. They are ways to keep your Angular app clean and streamline your code.    
+Angular allow you to lazy load each feature module in order to boost the performance of your app.
+Feature encapsulation using module allow you to replace each module without (or with just small changes) affect other module. It extend Angular application with capabilities from external libraries and relieved a future inclusion of new concepts like _micro-frontends_ or _monorepos_.
+
+Here's an example. 
+
+```ts
+
+// AddPeopleComponent
+@Component({
+  ...
+})
+export class AddPeopleComponent implements OnInit {
+  ...
+}
+
+// ListPeopleComponent
+@Component({
+  ...
+})
+export class ListPeopleComponent implements OnInit {
+  ...
+}
+
+// AppModule
+@NgModule({
+  ...
+  imports: []
+  declarations: [
+    AddPeopleComponent,
+    ListPeopleComponent
+    // Other components
+  ],
+  ...
+})
+export class AppModule {}
+```
+
+can be refactored to: 
+
+
+```ts
+
+// AddPeopleComponent
+@Component({
+  ...
+})
+export class AddPeopleComponent implements OnInit {
+  ...
+}
+
+// ListPeopleComponent
+@Component({
+  ...
+})
+export class ListPeopleComponent implements OnInit {
+  ...
+}
+
+// Introduce a new feature module
+@NgModule({
+  imports: [
+    CommonModule
+  ]
+  declarations: [
+    AddPeopleComponent,
+    ListPeopleComponent
+  ]
+})
+export class PeopleModule {}
+
+// AppModule
+@NgModule({
+  ...
+  imports: [
+    // import our new Module
+    PeopleModule
+  ]
+  declarations: [
+    // Other components
+  ],
+  ...
+})
+export class AppModule {}
+```
+
+or using lazy loading like described [here](/default/checklist/router/Z165VzV).
```

Comment:
I suggest => Alternatively, we can use lazy loading as described [here](...).

The only problem is that links might break for users if they don't have the default project anymore. We should address this in a different PR.